salt/beacons/service.py

```diff
@@ -72,21 +80,28 @@ def beacon(config):
         ret_dict = {}
         ret_dict[service] = {'running': __salt__['service.status'](service)}
 
+        # If no options is given to the service, we fall back to the defaults
+        # assign a False value to oncleanshutdown and onchangeonly. Those
+        # key:values are then added to the service dictionary.
+        if config[service] is None:
+            defaults = {
+                    'oncleanshutdown': False,
+                    'onchangeonly': False
+                    }
+            config[service] = defaults
+
         # We only want to report the nature of the shutdown
         # if the current running status is False
         # as well as if the config for the beacon asks for it
         if 'uncleanshutdown' in config[service] and not ret_dict[service]['running']:
             filename = config[service]['uncleanshutdown']
-            if os.path.exists(filename):
-                ret_dict[service]['shutdown'] = 'unclean'
-            else:
-                ret_dict[service]['shutdown'] = 'clean'
-
+            ret_dict[service]['uncleanshutdown'] = True if os.path.exists(filename) else False
+            
         if 'onchangeonly' in config[service] and config[service]['onchangeonly'] is True:
-            if service not in last_status:
-                last_status[service] = ''
+            if service not in LAST_STATUS:
+                LAST_STATUS[service] = ''
             if last_status[service] != ret_dict[service]:
```

Comment:
I absolutely did not, that's what I get for using the editor on GitHub.
Fixed it and ran it through pylint, seems good now.